src/seer.erl

```diff
@@ -88,26 +95,40 @@ get(Type, Name) when is_binary(Name) ->
 get(_Type, _Name) ->
     {error, metric_name_not_a_binary}.
 
+-spec histo_percentiles(non_neg_integer(),
+                        [histo_bucket()],
+                        [histo_percentile_index()]) -> #{histo_percentile() => histo_bucket_key()} | #{}.
 histo_percentiles(N, Buckets, PercentileIdxes) ->
     histo_percentiles(N, Buckets, PercentileIdxes, #{}).
 
+% TODO: find out why using #{histo_percentile() => histo_bucket_key()} | #{}
```

Comment:
@FrankBro maybe you can shed light on this?

Reply:
`#{}` is not `map()`. `#{}` is explicitly an empty map. Also keep in mind I have ANOTHER branch opened for some map bugs that existed but I don't think your code triggers it https://github.com/josefs/Gradualizer/pull/286

The code isnt finished and I hope I get more time to fix it in the next sprint or the one after.

Reply:
So I think because you return `Res`, which is a `#{}`, the type system has no proof that at this point, the map is empty. Check this https://github.com/josefs/Gradualizer/wiki/Map-types, the signature of `:=` vs `=>`. I think `Res` here should have type `#{histo_percentile() => histo_bucket_key()}` as well. Since it's optional, it CAN not be present and should unify.